src/textual/_styles_cache.py

```diff
@@ -52,6 +54,20 @@ def style_links(
     return segments
 
 
+@lru_cache(1024 * 8)
```

Comment:
When we paint widgets we tend to store a lot of blank strings for backgrounds. This caches those blank segments, more for memory than cpu.